### PR TITLE
Fix counter which is not updated in the loop

### PR DESCRIPTION
**Pull request by Egor18**
The counter `cnt` is not updated inside the loop here, so the condition `cnt<5` is always true.
The code has been this way for a long time, and I don't see why we should iterate only 5 times here, so instead of adding `i++`, I simply removed it. Correct me if I'm wrong.
(I found this issue with data-flow analyzer)

### CODE REVIEW

**Review by res0nance — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
LGTM. Looks to be a historical performance tweak

**Review by varyvol — APPROVED** [review]

File: core/src/main/java/hudson/model/AbstractProject.java [hunk 1]
```diff
@@ -553,17 +553,15 @@ private AbstractBuild getBuildForDeprecatedMethods() {
      * @return null if no such build exists.
      */
     public final R getSomeBuildWithWorkspace() {
-        int cnt=0;
-        for (R b = getLastBuild(); cnt<5 && b!=null; b=b.getPreviousBuild()) {
+        for (R b = getLastBuild(); b!=null; b=b.getPreviousBuild()) {
```
Comment (alecharp) [inline core/src/main/java/hudson/model/AbstractProject.java]:
```suggestion
        for (R b = getLastBuild(); b != null; b = b.getPreviousBuild()) {
```

File: core/src/main/java/hudson/model/AbstractProject.java [hunk 2]
```diff
@@ -553,17 +553,15 @@ private AbstractBuild getBuildForDeprecatedMethods() {
...
             FilePath ws = b.getWorkspace();
             if (ws!=null)   return b;
         }
         return null;
     }
 
     private R getSomeBuildWithExistingWorkspace() throws IOException, InterruptedException {
-        int cnt=0;
-        for (R b = getLastBuild(); cnt<5 && b!=null; b=b.getPreviousBuild()) {
+        for (R b = getLastBuild(); b!=null; b=b.getPreviousBuild()) {
```
Comment (alecharp) [inline core/src/main/java/hudson/model/AbstractProject.java]:
```suggestion
        for (R b = getLastBuild(); b != null; b = b.getPreviousBuild()) {
```

**Review by alecharp — CHANGES_REQUESTED** [review]
Just some suggestion to add spaces but it's good otherwise. Thanks!

**Comment by oleg-nenashev** [comment]
will merge tomorrow if no negative feedback

**Review by alecharp — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Thanks @Egor18 !

**Comment by daniel-beck** [comment]
> I don't see why we should iterate only 5 times here

Probably due to lazy build loading -- if builds are too old, don't force them to be loaded. Same as why weather or estimated build duration only consider the five latest builds.

Anyone willing to amend this?

**Pull request merged**